src/core/server/http/cookie_session_storage.ts

```diff
@@ -98,15 +104,35 @@ export async function createCookieSessionStorageFactory<T>(
   cookieOptions: SessionStorageCookieOptions<T>,
   basePath?: string
 ): Promise<SessionStorageFactory<T>> {
+  function clearInvalidCookie(
+    req: Request | undefined,
+    path: string = basePath || '/',
+    isSecure: boolean = cookieOptions.isSecure
```

Comment:
Assume that, if a cookie's session value does not contain the `path` and `isSecure` attributes, then it was most likely created with the current configuration (and an older version of Kibana).

There is still a corner case where a user may have changed the Kibana config _and_ upgraded to a version with these changes at the same time, unfortunately we can't do anything about that (and users would experience the same infinite redirect issues that they do today).